### PR TITLE
♻️ CI: only run tests jobs on path changes in pull requests

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

This PR is part of an ongoing refactoring of the CI workflow file used with Github Workflow (the actual Github CI).
With this PR:
- in Pull Requests only the jobs which are detected necessary will be run 
  - example A: if some Javascript files are changed, only the frontend unit tests, build of new images and the system tests will be run
  - example B: if some python files are changed in service director-v2, then only the unit-test and integration-tests of director-v2, as well as python linter and system tests will be run

Some changes in the Repository settings will be done:
- currently all the jobs are a required status
- After this PR: only the jobs named ```unit-tests```, ```integration-tests``` and ```system-tests``` will be set as required
- These 3 *aggregator jobs* will be green only if their dependencies were run successfully or skipped thus allowing PRs with skipped jobs to be merged

NOTE: on the main/master branch **ALL** tests will always be run to prevent any missed error from slipping in!

Bonus:
- removed the code climate coverage steps/job as these were unused and anyway not functional. The usual code climate job still is running (this only is about code coverage)

### CODE REVIEW

File: .github/workflows/ci-testing-deploy.yml [hunk 1]
```diff
@@ -1178,21 +1149,45 @@ jobs:
...
+        unit-test-webserver-03,
+      ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: all unit tests were run or skipped
+        run: echo "all unit tests were run or skipped"
 
   integration-test-webserver-01:
+    needs: [changes, build-test-images]
+    if: ${{ needs.changes.outputs.anything-py == 'true' || github.event_name == 'push' }}
```
Comment (GitHK) [inline .github/workflows/ci-testing-deploy.yml:1188]: For integration testing, it is always triggered if some python code changes? Is my interpretation correct?

**Review by GitHK — APPROVED** [review]
Very nice addition!

Question: let's say I slip up and merge something that breaks the build on master. At this point I expect no images to be pushed to dockerhub, right? So it will work as before, nothing will be broken, but I will know if I've broken something only when running against master. I guess we expect this only for edge cases.

File: .github/workflows/ci-testing-deploy.yml (hunk 1 above)
Comment (sanderegg) [reply .github/workflows/ci-testing-deploy.yml:1188]: that is correct. it could be more granular, but I thought this would need too much maintenance as it would have to be in sync with the package dependencies.

**Comment by sanderegg** [comment]
> Very nice addition!
> 
> Question: let's say I slip up and merge something that breaks the build on master. At this point I expect no images to be pushed to dockerhub, right? So it will work as before, nothing will be broken, but I will know if I've broken something only when running against master. I guess we expect this only for edge cases.

@GitHK , you are right on master all the jobs will always be run. so this should prevent any such event.

**Review by mrnicegyu11 — APPROVED** [review]
This looks like a promising route to run the tests more rapid, thanks a lot and good luck!

**Review by pcrespov — APPROVED** [review]
Excellent! this was a long-awaited improvement in the CI! 🎉
